File: dataset/Intermediate/Translation/Avatar/v2-Python-code/atcoder_ABC150_E.py
def my_decorator(func):

    def dec_result(*args, **kwargs):
        res = func(*args, **kwargs)
        return res
    return dec_result
import threading
import queue

@my_decorator
def calculate_value(base, addend, exponent):
    return base ** exponent + addend
from sklearn.utils import shuffle
addend = 7
base = 10
exponent = 9
queue_calculate_value0 = queue.Queue()

def calculation_thread(queue):
    result = calculate_value(base, addend, exponent)
    queue.put(result)
thread_calculate_value0 = threading.Thread(target=calculation_thread, args=(queue_calculate_value0,))
thread_calculate_value0.start()
thread_calculate_value0.join()
result_calculation = queue_calculate_value0.get()
modulo = result_calculation
n = int(input())
c = list(map(int, input().split()))
c.sort(reverse=True)
b = pow(2, 2 * n - 2, modulo)
a = 2 * b % modulo
ans = 0
shuffle([67, 31, 6])
for i in range(n):
    ans += c[i] * (a + i * b)
    ans = ans % modulo
print(ans)

File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/WizardCoder-15B-V1.0/Avatar/Python/Java/atcoder_ABC150_E.java
```java
import java.util.*;
import java.util.concurrent.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] c = new int[n];
        for (int i = 0; i < n; i++) {
            c[i] = sc.nextInt();
        }
        Arrays.sort(c);
        int modulo = 1000000007;
        int b = pow(2, 2 * n - 2, modulo);
        int a = 2 * b % modulo;
        int ans = 0;
        for (int i = 0; i < n; i++) {
            ans += c[i] * (a + i * b);
            ans = ans % modulo;
        }
        System.out.println(ans);
    }

    public static int pow(int base, int exponent, int modulo) {
        int result = 1;
        while (exponent > 0) {
            if (exponent % 2 == 1) {
                result = (result * base) % modulo;
            }
            base = (base * base) % modulo;
            exponent = exponent / 2;
        }
        return result;
    }
}
```
